core/src/test/scala/unit/kafka/integration/MetricsDuringTopicCreationDeletionTest.scala

```diff
@@ -0,0 +1,163 @@
+  
+  override def generateConfigs() = TestUtils.createBrokerConfigs(nodesNum, zkConnect)
+    .map(KafkaConfig.fromProps(_, overridingProps))
+
+  @Before
+  override def setUp {
+    // Do some Metrics Registry cleanup by removing the metrics that this test checks. 
+    // This is a test workaround to the issue that prior harness runs may have left a populated registry.
+    // see https://issues.apache.org/jira/browse/KAFKA-4605
+    for (m <- (testedMetrics)) {
+        Metrics.defaultRegistry.allMetrics.asScala
+        .filterKeys(k => k.getName.endsWith(m))
+        .headOption match {
+           case Some(e) => Metrics.defaultRegistry.removeMetric(e._1)
+           case None =>
+        }
+    }
+    
+    super.setUp
+  }
+
+  /*
+   * checking all metrics we care in a single test is faster though it would be more elegant to have 3 @Test methods
+   */
+  @Test
+  def testMetricsDuringTopicCreateDelete() {
+
+    // For UnderReplicatedPartitions, because of https://issues.apache.org/jira/browse/KAFKA-4605
+    // we can't access the metrics value of each server. So instead we directly invoke the method 
+    // replicaManager.underReplicatedPartitionCount() that defines the metrics value.
+    @volatile var underReplicatedPartitionCount = 0
+
+    // For OfflinePartitionsCount and PreferredReplicaImbalanceCount even with https://issues.apache.org/jira/browse/KAFKA-4605
+    // the test has worked reliably because the metric that gets triggered is the one generated by the first started server (controller)
+    val offlinePartitionsCountGauge = getGauge("OfflinePartitionsCount")
+    @volatile var offlinePartitionsCount = offlinePartitionsCountGauge.value
+    assert(offlinePartitionsCount == 0)
+
+    val preferredReplicaImbalanceCountGauge = getGauge("PreferredReplicaImbalanceCount")
+    @volatile var preferredReplicaImbalanceCount = preferredReplicaImbalanceCountGauge.value
+    assert(preferredReplicaImbalanceCount == 0)
+
+    // Thread checking the metric continuously
+    running = true
+    val thread = new Thread(new Runnable {
+      def run() {
+        while (running && !Thread.interrupted) {
+          for ( s <- servers if running) {
+            underReplicatedPartitionCount = s.replicaManager.underReplicatedPartitionCount
+            if (underReplicatedPartitionCount > 0) {
+              running = false
+            }
+          }
+
+          preferredReplicaImbalanceCount = preferredReplicaImbalanceCountGauge.value
+          if (preferredReplicaImbalanceCount > 0) {
+             running = false
+          }
+
+          offlinePartitionsCount = offlinePartitionsCountGauge.value
+          if (offlinePartitionsCount > 0) {
+             running = false
+          }
+        }
+      }
+    })
+    thread.start
+
+    // breakable loop that creates and deletes topics
+    createDeleteTopics()
+
+    // if the thread checking the gauge is still run, stop it
+    running = false;
+    thread.interrupt
```

Comment:
Do we need to interrupt? There is no blocking operation in the thread. So it seems just checking running is enough.